### PR TITLE
CausedBy IError

### PR DESCRIPTION
**Pull request by eltoncezar**
Hi,

Is there a reason why `CausedBy` only allows passing `Error` and not `IError`?

```csharp
var result1 = Result.Fail("Some error");
var result2 = Result.Fail(new Error("Another error").CausedBy(result1.Errors));
```

### CODE REVIEW

**Comment by altmann** [comment]
No there is no good reason. :) Thanks for the pr - I will review it in the next days.

File: src/FluentResults/Reasons/Error.cs [hunk 1]
```diff
@@ -67,6 +67,18 @@ public Error CausedBy(Error error)
             return this;
         }
 
+        /// <summary>
+        /// Set the root cause of the error
+        /// </summary>
+        public Error CausedBy(IError error)
```
Comment (altmann) [inline src/FluentResults/Reasons/Error.cs:73]: Please don't add new methods. Only change the existing method with Error error to IError error.

File: src/FluentResults/Reasons/Error.cs [hunk 2]
```diff
@@ -112,6 +124,18 @@ public Error CausedBy(IEnumerable<Error> errors)
             return this;
         }
 
+        /// <summary>
+        /// Set the root cause of the error
+        /// </summary>
+        public Error CausedBy(IEnumerable<IError> errors)
```
Comment (altmann) [inline src/FluentResults/Reasons/Error.cs:130]: Please don't add new methods. Only change the existing method with IEnumerable<Error> errors to IEnumerable<IError> errors.

File: src/FluentResults/Reasons/Error.cs [hunk 3]
```diff
@@ -24,7 +24,7 @@ public class Error : IError
         /// Get the reasons of an error
         /// </summary>
         public List<IError> Reasons { get; }
-        
+
         protected Error()
```
Comment (altmann) [inline src/FluentResults/Reasons/Error.cs:28]: There is a constructor with

```
public Error(string message, Error causedBy)
            : this(message)
```

Please change it also to an interface

```
public Error(string message, IError causedBy)
            : this(message)
```

File: src/FluentResults/Reasons/Error.cs (hunk 1 above)
Comment (eltoncezar) [reply src/FluentResults/Reasons/Error.cs:61]: Done.

File: src/FluentResults/Reasons/Error.cs (hunk 2 above)
Comment (eltoncezar) [reply src/FluentResults/Reasons/Error.cs:106]: Done.

File: src/FluentResults/Reasons/Error.cs (hunk 3 above)
Comment (eltoncezar) [reply src/FluentResults/Reasons/Error.cs:28]: Done.

**Comment by altmann** [comment]
A new version is released https://www.nuget.org/packages/FluentResults/3.6.0

Thanks again for your time!

**Pull request merged**